src/lib.rs

```diff
@@ -155,6 +157,32 @@ pub fn build_from_docker(
     Ok((file_output, measurements))
 }
 
+/// Creates new enclave name
+///
+/// Requests the names of all running instances and checks the
+/// occurrence of the chosen name for the new enclave.
+pub fn new_enclave_name(run_args: RunEnclavesArgs, names: Vec<String>) -> NitroCliResult<String> {
+    let enclave_name = match run_args.enclave_name {
+        Some(enclave_name) => enclave_name,
+        None => {
+            // Get name of EIF file from path eg. path/to/eif/hello.eif -> hello
+            let path_split: Vec<&str> = run_args.eif_path.split(&['/', '.'][..]).collect();
```

Comment:
We should not assume that all the EIF paths have the `.eif` extension as we do not enforce this rule anywhere.

Reply:
Solved to treat the case of no extension present.